internal/distributed/proxy/httpserver/utils.go

```diff
@@ -1091,6 +1092,26 @@ func serializeByteVectors(vectorStr string, dataType schemapb.DataType, dimensio
 	return values, nil
 }
 
+func serializeFp16OrBf16Vectors(vectorStr string, dataType schemapb.DataType, dimension int64, serializeFunc func([]float32) []byte) ([][]byte, error) {
+	// try to unmarshal as [][]float32 first to make sure `[[3, 3]]` is [][]float32 instead of [][]byte
+	fp32Values := make([][]float32, 0)
+	err := json.Unmarshal([]byte(vectorStr), &fp32Values)
+	if err == nil {
+		values := make([][]byte, 0)
+		for _, vectorArray := range fp32Values {
+			if int64(len(vectorArray)) != dimension {
+				vecStr, _ := json.MarshalToString(vectorArray)
```

Comment:
the vecStr seems only used to report err? It is recommended not to use marshal one more time to get vecStr, but to pass in other valid error information, such as field name, to prompt the user to pass in the correct data.